modules/ssh/init.zsh

```diff
@@ -28,6 +33,14 @@ if [[ ! -S "$SSH_AUTH_SOCK" ]]; then
   if ! ps -U "$LOGNAME" -o pid,ucomm | grep -q -- "${SSH_AGENT_PID:--1} ssh-agent"; then
     eval "$(ssh-agent | sed '/^echo /d' | tee "$_ssh_agent_env")"
   fi
+elif [[ "$SSH_AUTH_SOCK" == "$(launchctl getenv SSH_AUTH_SOCK)" ]]; then
```

Comment:
SSH_AUTH_SOCK is an environment variable that gets set on starting your login session by launchd.  The system default behavior is to have this set to a launchd socket that resides in /tmp and captures the incoming connections to the system default ssh-agent and then starts it on demand and uses socket redirection to send these connections to it.

In the 'ssh' module prior to my patch, it was merely checking to see if $SSH_AUTH_SOCK was existed and was a socket.  Since this will always be true, it never causes the ssh-agent to get started on OS X.  After my modification, it checks to see if it's matching the value launchctl will set on login, and if so, continues setting up the connection.  During the ssh-agent startup process SSH_AUTH_SOCK gets changed.